### ISSUE TITLE
Issue with predict_classes/predict_prob

### ISSUE THREAD

**Issue by FelixZhao123** [rstudio/keras#390]
I have a ANN model trained and when i tried to make predictions, neither predict_classes nor predict_proba works.
`library(tidyverse)
library(keras)### deep learning training and modeling
library(lime) ## model explanation
library(recipes) ### data preprocessing
library(rsample)  ### dataset sampling
library(yardstick)  ## mdoel performance metrics
library(corrr)   ## correlation analysis
library(tidyquant)

### import dataset
churn_data_raw <- read_csv("Telco Customer Churn.csv",col_names = TRUE)

### remove na's and de-select unneccessary variables
churn_data_raw <- churn_data_raw%>%
  select(-customerID)%>%
  drop_na()%>%
  select(Churn,everything())

### resampling and splitting dataset into train_test using "rsampling" package

set.seed(1002)
train_test_split <- initial_split(churn_data_raw,prop = 0.8)
train_test_split
### the total number of observations is 7032

train_tbl <- training(train_test_split)
test_tbl <- testing(train_test_split)

#############################################
### preprocessing data with "recipe" package

rec_object <- recipe(Churn~.,data=train_tbl)%>%
  step_discretize(tenure,options = list(cuts = 6))%>%
  step_log(TotalCharges)%>%
  step_dummy(all_nominal(),-all_outcomes())%>%
  step_center(all_predictors(),-all_outcomes())%>%
  step_scale(all_predictors(),-all_outcomes())%>%
  prep(data=train_tbl)

### now "bake" data with the receipe we created

x_train_tbl <- bake(rec_object,newdata = train_tbl)
x_test_tbl <- bake(rec_object,newdata = test_tbl)

### pull out the target and drop target column from original training and testing datasets
y_train_vec <- ifelse(pull(train_tbl,Churn)=="Yes",1,0)
y_test_vec <- ifelse(pull(test_tbl,Churn)=="Yes",1,0)

### remove target from column from preprocessed training/testing data frames
x_train_tbl <- x_train_tbl%>%
  select(-Churn)
x_test_tbl <- x_test_tbl%>%
  select(-Churn)

keras_model_1 <- keras_model_sequential()

keras_model_1%>%
  ## first fully connected layer
  layer_dense(
    units = 16,
    kernel_initializer = "uniform",
    activation = "relu",
    input_shape = ncol(x_train_tbl)
  )%>%
  
##dropout to prevent overfitting
  layer_dropout(rate = 0.1)%>%
  ### add second hidden layer
  layer_dense(
    units = 16,
    kernel_initializer = "uniform",
    activation = "relu"
  )%>%
  layer_dense(units = 5,
              activation = 'relu',
              kernel_initializer = 'normal')%>%
  ##add output layer
  layer_dense(
    units = 1,
    kernel_initializer="uniform",
    activation = "sigmoid"
  )%>%
  compile(
    optimizer = "adam",
    loss = "binary_crossentropy",
    metrics = c('accuracy')
  )

model_1_fit <- fit(
  object = keras_model_1,
  x = as.matrix(x_train_tbl),
  y = y_train_vec,
  batch_size = 50,
  epochs = 200,
  validation_split = 0.2
)

model_1_fit

### type 1: predict classes

yhat_keras_class_vec <- predict_classes(object = model_1_fit,x=as.matrix(x_test_tbl))%>%
  as.vector()
`
`Error in do.call(object$predict_classes, args) : 
  'what' must be a function or character string`
`sessionInfo()`
`R version 3.4.1 (2017-06-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default`
`attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] yardstick_0.0.1            tidyquant_0.5.4           
 [3] quantmod_0.4-13            TTR_0.23-3                
 [5] PerformanceAnalytics_1.5.2 xts_0.10-2                
 [7] zoo_1.8-1                  lubridate_1.7.3           
 [9] corrr_0.2.1                rsample_0.0.2             
[11] recipes_0.1.2              broom_0.4.4               
[13] lime_0.4.0                 keras_2.1.6               
[15] forcats_0.2.0              stringr_1.3.0             
[17] dplyr_0.7.4                purrr_0.2.4               
[19] readr_1.1.1                tidyr_0.8.0               
[21] tibble_1.3.4               ggplot2_2.2.1             
[23] tidyverse_1.2.1           

loaded via a namespace (and not attached):
 [1] nlme_3.1-131       dimRed_0.1.0       httr_1.3.1        
 [4] tools_3.4.1        R6_2.2.2           rpart_4.1-11      
 [7] lazyeval_0.2.0     colorspace_1.3-2   nnet_7.3-12       
[10] tidyselect_0.2.4   mnormt_1.5-5       curl_3.2          
[13] compiler_3.4.1     glmnet_2.0-13      cli_1.0.0         
[16] rvest_0.3.2        xml2_1.1.1         scales_0.5.0      
[19] sfsmisc_1.1-2      DEoptimR_1.0-8     psych_1.8.3.3     
[22] robustbase_0.92-8  quadprog_1.5-5     tfruns_1.3        
[25] digest_0.6.15      foreign_0.8-69     stringdist_0.9.4.7
[28] base64enc_0.1-3    pkgconfig_2.0.1    htmltools_0.3.6   
[31] htmlwidgets_0.9    rlang_0.2.0        readxl_1.0.0      
[34] ddalpha_1.3.1.1    MLmetrics_1.1.1    rstudioapi_0.7    
[37] shiny_1.0.5        bindr_0.1.1        jsonlite_1.5      
[40] tensorflow_1.5     magrittr_1.5       Matrix_1.2-10     
[43] Quandl_2.8.0       Rcpp_0.12.16       munsell_0.4.3     
[46] reticulate_1.6     pROC_1.11.0        stringi_1.1.7     
[49] whisker_0.3-2      yaml_2.1.14        MASS_7.3-47       
[52] plyr_1.8.4         grid_3.4.1         parallel_3.4.1    
[55] crayon_1.3.4       lattice_0.20-35    haven_1.1.0       
[58] splines_3.4.1      hms_0.3            magick_1.8        
[61] zeallot_0.1.0      knitr_1.19         reshape2_1.4.3    
[64] codetools_0.2-15   CVST_0.2-1         glue_1.1.1        
[67] modelr_0.1.1       httpuv_1.3.6.2     foreach_1.4.4     
[70] cellranger_1.1.0   gtable_0.2.0       kernlab_0.9-25    
[73] assertthat_0.2.0   DRR_0.0.3          gower_0.1.2       
[76] mime_0.5           prodlim_1.6.1      xtable_1.8-2      
[79] class_7.3-14       survival_2.41-3    timeDate_3043.102 
[82] RcppRoll_0.2.2     shinythemes_1.1.1  iterators_1.0.9   
[85] bindrcpp_0.2       lava_1.5           ipred_0.9-6      `
